crates/turbopack-core/src/resolve/alias_map.rs

```diff
@@ -192,14 +193,16 @@ impl<T> AliasMap<T> {
     /// Looks up a request in the alias map.
     ///
     /// Returns an iterator to all the matching aliases.
-    pub fn lookup<'a>(&'a self, request: &'a str) -> AliasMapLookupIterator<'a, T>
+    pub fn lookup<'a>(&'a self, request: &'a Pattern) -> AliasMapLookupIterator<'a, T>
     where
         T: Debug,
     {
         // Invariant: prefixes should be sorted by increasing length (base lengths),
         // according to PATTERN_KEY_COMPARE. Since we're using a prefix tree, this is
         // the default behavior of the common prefix iterator.
-        let common_prefixes = self.map.common_prefixes(request.as_bytes());
+        let common_prefixes = self
+            .map
+            .common_prefixes(request.constant_prefix().as_bytes());
```

Comment:
This will lead to also returning non-matching results. e. g. Pattern could be `"a" | "b"` which has no constant_prefix so it would return `"c"` too.

You probably want:
* Do multiple lookups internally when you have Alternatives
* Filter the results by matching with the Pattern

Reply:
Looks like this doesn't happen, `map.common_prefixes("")` is just empty. That does align with example on https://docs.rs/patricia_tree/latest/patricia_tree/map/type.PatriciaMap.html#method.common_prefixes

But still, `map.lookup(Pattern::Alternative(...))` currently never returns anything even if some/all alternatives match. And even then, just returning `enum AliasMatch<R> {Exact(R), Replaced(R)}` is wrong anyway because some might be exact and some might be replaced matches